qa/workunits/fs/test-volumes.sh

```diff
@@ -28,10 +22,83 @@ ceph fs volume create bar
 ceph fs volume ls | grep bar
 ceph fs volume rm bar
 
+### basic subvolume tests
+
 # subvolumes on $EXISTING
 ceph fs subvolume create $EXISTING sub1
 ceph fs subvolume create $EXISTING sub2
+# make sure it exists
+ceph fs subvolume getpath $EXISTING sub1
+ceph fs subvolume getpath $EXISTING sub2
+
+# creating an already existing subvolume is idempotent
+ceph fs subvolume create $EXISTING sub1
+
+# remove subvolumes and verify
+ceph fs subvolume rm $EXISTING sub1
 ceph fs subvolume rm $EXISTING sub2
+# make sure its removed. note that this is a negative test
+ceph fs subvolume getpath $EXISTING sub1 && exit 1
+ceph fs subvolume getpath $EXISTING sub2 && exit 1
+
+# deleting an already deleted subvolume
+ceph fs subvolume rm $EXISTING sub1 && exit 1
+ceph fs subvolume rm $EXISTING sub1 --force
+
+# create and delete subvolumes w/ same subvol name as above
+# this would test subvole purge queue when that's implemented
+# (currently, deleting subvolumes is an synchronous operation)
+ceph fs subvolume create $EXISTING sub1
+ceph fs subvolume create $EXISTING sub2
+ceph fs subvolume rm $EXISTING sub1
+ceph fs subvolume rm $EXISTING sub2
+
+### subvolume group tests
+
+# create a subvolume in an non-existing subvolume group should fail
+ceph fs subvolume create $EXISTING sub3 --group_name non_existent_group && exit 1
+
+# create a subvolume group
+ceph fs subvolumegroup create $EXISTING group_1
+
+# create subvolumes in the subvolume group
+ceph fs subvolume create $EXISTING sub3 --group_name group_1
+ceph fs subvolume create $EXISTING sub4 --group_name group_1
+
+# snap the subvolume group
+ceph fs subvolumegroup snapshot create $EXISTING group_1 group_snap_1
+# remove subvolume group snaphot
+ceph fs subvolumegroup snapshot rm $EXISTING group_1 group_snap_1
```

Comment:
Test removal of non existent subvolumegroup snapshot with and without --force flag?